### DISCUSSION TITLE
Generate a SQL schema file

### DISCUSSION THREAD

**Discussion by plutino** [ariga/atlas#1643] [Q&A] [answered]
Is there any way to generate a SQL schema file from the existing versioned migrations?  We would like to be able to create a test db server quickly with an empty schema, and the easiest way would be generate the current SQL schema and pass it to PostgreSQL server during start.

**Comment by rotemtam** [comment, marked as answer]
Hey @plutino 

Thanks for reaching out. It's possible to use "schema inspect" on a migration dir. by using SQL formatting you can get the "sum" schema:

atlas schema inspect -u "file://path/to/dir" --dev-url docker://postgres/15 --format "{{ sql . }}"

Notice that if you use "schema apply" you can directly apply this schema to your target db.

**Reply by plutino** [reply]
This is very handy, thanks @rotemtam.  Is there any way to avoid using a dev db?  Ideally we only need to generate the schema file when building a docker image for a testing db server.  Requiring the dev db means to have to run a db server when building the image.

**Reply by plutino** [reply]
One issue we noticed is that the generated SQL file does not include Postgres extensions or functions.  We also noticed that the initial import from an existing database does not include extensions and functions either.  Could this be something that is missed by the Postgres driver?
